[FSDP][BE] get_full_params uses summon_full_params

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73242
* #73314

Can use summon_full_params instead.

Differential Revision: [D34399789](https://our.internmc.facebook.com/intern/diff/D34399789/)

rohan-varma committed:
[FSDP][BE] remove get_full_params() from test code

Can use summon_full_params instead.

Differential Revision: [D34399789](https://our.internmc.facebook.com/intern/diff/D34399789/)

[ghstack-poisoned]

zhaojuanmao commented on torch/testing/_internal/common_fsdp.py:
```diff
@@ -32,22 +33,6 @@ class FSDPInitMode(Enum):
-# also automatically move the parameters to GPU, due to _rebuild_full_params
-# call.
-def get_full_params(model, recurse=True):
-    if recurse:
-        # get all params for any nested FSDP instances.
-        for module in model.modules():
-            if isinstance(module, FullyShardedDataParallel):
-                get_full_params(module, recurse=False)
-    else:
-        torch.cuda.synchronize()
-        model._rebuild_full_params()
-        torch.cuda.synchronize()
-        if model.module.flat_param is not None:
-            model.module._unflatten_params()
-
```
just replace this with following? so that other places do not need to change?

            with model._summon_full_params():
                return deepcopy(list(model.parameters()))

rohan-varma committed:
Update on "[FSDP][BE] remove get_full_params() from test code"


Can use summon_full_params instead.

Differential Revision: [D34399789](https://our.internmc.facebook.com/intern/diff/D34399789/)

[ghstack-poisoned]

rohan-varma commented:
@zhaojuanmao Good call, just changed it

rohan-varma committed:
Update on "[FSDP][BE] get_full_params uses summon_full_params"


Can use summon_full_params instead.

Differential Revision: [D34399789](https://our.internmc.facebook.com/intern/diff/D34399789/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP][BE] get_full_params uses summon_full_params"


Can use summon_full_params instead.

Differential Revision: [D34399789](https://our.internmc.facebook.com/intern/diff/D34399789/)

[ghstack-poisoned]
